optaplanner-docs/src/main/asciidoc/PlannerConfiguration/PlannerConfiguration-chapter.adoc

```diff
@@ -136,27 +135,29 @@ These `$$*$$Config` classes are the Java representation of the XML format.
 They build the runtime components (of the package namespace ``org.optaplanner.core.impl``)
 and assemble them into an efficient ``Solver``.
 
-[IMPORTANT]
+[NOTE]
 ====
-The `SolverFactory` is only thread-safe after it's configured.
-So the `getSolverConfig()` method is not thread-safe.
-To configure a `SolverFactory` dynamically for each user request, build a basic `SolverFactory` during initialization
-and clone it with the `cloneSolverFactory()` method for a user request:
+Don't modify a `SolverConfig` after creating a `SolverFactory`,
+because it currently breaks the thread-safety of the `SolverFactory`.
+To configure a `SolverFactory` dynamically for each user request,
+build a template `SolverConfig` during initialization
+and copy it with the copy constructor for each user request:
```

Comment:
This is indeed surprising. Can we make the defensive copy ourselves and not bother the user with such details?

Reply:
Yeah, +1 on the defensive copy. I don't want to be thinking about when I can or can not modify the config.